From e66e40397e25f7ff978ef3f203f82e373b21d396 Mon Sep 17 00:00:00 2001
From: wwidjaja <w.widjaja@intel.com>
Date: Wed, 19 Feb 2020 15:32:29 +0800
Subject: [PATCH] PONRTSYS-4533 - configure TEP DDR region permission

  set NGI permission for TEP DDR protection region to 4kec only
  DDR region (0x3E000000~0x3FFFFFFF) 32 MB

  NOTE: to match with uboot FMT1 programming
        + CONFIG_PRX300_TOS_32M
        + CONFIG_PRX300_DDR_512M
---
 arch/mips/lantiq/grx500_bootcore/ngi_prx300.c      | 7 +++++--
 arch/mips/lantiq/grx500_bootcore/ngi_prx300_ssx1.h | 1 +
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
index 16cac8d6855e..11c1715a8b81 100644
--- a/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
+++ b/arch/mips/lantiq/grx500_bootcore/ngi_prx300.c
@@ -87,10 +87,13 @@ void set_permission(void)
 			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_0, ALL_CPUS);
 	permission_wr(ssx1_base,
 			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_0, ALL_CPUS);
+	/* protect DDR region 0x3E000000~0x40000000 32MB */
 	permission_wr(ssx1_base,
-			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_1, ALL_CPUS);
+			REG0_TDDR_PM_TDDR_PM_ADDR_MATCH_1, 0x3E000081);
 	permission_wr(ssx1_base,
-			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_1, ALL_CPUS);
+			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_1, M4KEC_ONLY);
+	permission_wr(ssx1_base,
+			REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_1, M4KEC_ONLY);
 	permission_wr(ssx1_base,
 			REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_2, ALL_CPUS);
 	permission_wr(ssx1_base,
diff --git a/arch/mips/lantiq/grx500_bootcore/ngi_prx300_ssx1.h b/arch/mips/lantiq/grx500_bootcore/ngi_prx300_ssx1.h
index 4b5d25b35c4e..49b5f11139c5 100644
--- a/arch/mips/lantiq/grx500_bootcore/ngi_prx300_ssx1.h
+++ b/arch/mips/lantiq/grx500_bootcore/ngi_prx300_ssx1.h
@@ -30,6 +30,7 @@
 #define REG0_TEX04_PM_TEX04_PM_WRITE_PERMISSION_0		0x84058
 #define REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_0			0x84450
 #define REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_0			0x84458
+#define REG0_TDDR_PM_TDDR_PM_ADDR_MATCH_1				0x84460
 #define REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_1			0x84470
 #define REG0_TDDR_PM_TDDR_PM_WRITE_PERMISSION_1			0x84478
 #define REG0_TDDR_PM_TDDR_PM_READ_PERMISSION_2			0x84490
